src/compute/src/sink/materialized_view_v2.rs

```diff
@@ -33,5 +155,875 @@ pub(super) fn persist_sink<S>(
+fn persist_source<S>(
+    scope: &mut S,
+    sink_id: GlobalId,
+    target: CollectionMetadata,
+    compute_state: &ComputeState,
+    start_signal: StartSignal,
+) -> (PersistStreams<S>, Box<dyn Any>)
+where
+    S: Scope<Timestamp = Timestamp>,
+{
+    // There is no guarantee that the sink as-of is beyond the persist shard's since. If it isn't,
+    // instantiating a `persist_source` with it would panic. So instead we leave it to
+    // `persist_source` to select an appropriate as-of. We only care about times beyond the current
+    // shard upper anyway.
+    //
+    // TODO(teskje): Ideally we would select the as-of as `join(sink_as_of, since, upper)`, to
+    // allow `persist_source` to omit as much historical detail as possible. However, we don't know
+    // the shard frontiers and we cannot get them here as that requires an `async` context. We
+    // should consider extending the `persist_source` API to allow as-of selection based on the
+    // shard's current frontiers.
+    let as_of = None;
+
+    let until = Antichain::new();
+    let map_filter_project = None;
+
+    let (ok_stream, err_stream, token) = mz_storage_operators::persist_source::persist_source(
+        scope,
+        sink_id,
+        Arc::clone(&compute_state.persist_clients),
+        &compute_state.txns_ctx,
+        &compute_state.worker_config,
+        target,
+        as_of,
+        SnapshotMode::Include,
+        until,
+        map_filter_project,
+        compute_state.dataflow_max_inflight_bytes(),
+        start_signal,
+        |error| panic!("compute_persist_sink: {error}"),
+    );
+
+    let streams = OkErr::new(ok_stream, err_stream);
+    let token = Box::new(token);
+    (streams, token)
+}
+
+/// A description for a batch of updates to be written.
+///
+/// Batch descriptions are produced by the `mint` operator and consumed by the `write` and `append`
+/// operators, where they inform which batches should be written or appended, respectively.
+#[derive(Clone, Serialize, Deserialize)]
+struct BatchDescription {
+    lower: Antichain<Timestamp>,
+    upper: Antichain<Timestamp>,
+}
+
+impl BatchDescription {
+    fn new(lower: Antichain<Timestamp>, upper: Antichain<Timestamp>) -> Self {
+        debug_assert!(PartialOrder::less_than(&lower, &upper));
+        Self { lower, upper }
+    }
+}
+
+impl std::fmt::Debug for BatchDescription {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "({:?}, {:?})",
+            self.lower.elements(),
+            self.upper.elements()
+        )
+    }
+}
+
+/// Implementation of the `mint` operator.
+mod mint {
+    use super::*;
+
+    pub fn render<S>(
+        name: String,
+        persist_api: PersistApi,
+        as_of: Antichain<Timestamp>,
+        active_worker_id: usize,
+        desired: &DesiredStreams<S>,
+    ) -> (DesiredStreams<S>, DescsStream<S>, Box<dyn Any>)
```

Comment:
nit/optional: I'd either return the tokens as `PressOnDropButton` or `Rc<dyn Any>`. The former avoids the `Box` alloc or the latter is directly usable with our render stuff

Reply:
Changed all the tokens to `PressOnDropButton`s.